### DISCUSSION TITLE
Alignment and germline gene inference on pre-processed data

### DISCUSSION THREAD

**Discussion by giulioisac** [milaboratory/mixcr#1669] [Q&A] [answered]
Hello!

I have access to IgH pre-processed FASTA files obtained by running a standard pRESTO pipeline. The files contain already clustered/collapsed clonotypes. I would only need to perform an alignment to extract VDJ genes, CDR3 and infer the germline genes. Would that be possible within MiXCR? Thank you!

Best,
Giulio

**Comment by mizraelson** [comment, marked as answer]
Hi, yes, MiXCR can work with FASTA files.

If you only want to align the reads, without clustering them and correcting errors, the command would be

```
mixcr align --preset generic-amplicon \
    --species hsa \
    --rna \
    --floating-left-alignment-boundary \
    --floating-right-alignment-boundary C \
      input.fasta \
      result.vdjca
```

And then export those alignments with:

```
mixcr exportAlignments result.vdjca result.tsv
```

But if possible it is always better to process the raw FASTQ files with MiXCR.

**Reply by giulioisac** [reply]
That is super helpful thank you! And is it also possible to infer the germline genes using the 'mixcr findAlleles' command?

**Reply by mizraelson** [reply]
Running `findAlleles` requires `.clns` files that are obtained during clonotype assembly with MiXCR. You can try running the following command:


```
mixcr analyze generic-amplicon \
    --species hsa \
    --rna \
    --floating-left-alignment-boundary \
    --floating-right-alignment-boundary C \
    --assemble-clonotypes-by VDJRegion \
      input.fasta \
      result
```

Then, use the resulting .clns files for `findAlleles`. Note that I added `--assemble-clonotypes-by VDJRegion`, which requires your sequences to cover the FR1 to FR4 regions. Also, MiXCR will assemble clones from your sequences, and some of them might be collapsed, depending on the data.

**Reply by giulioisac** [reply]
That looks like a good workaround. Finally I won't need the .clns files for anything else, but I can use them to get the alleles and then do the alignment above with the right custom alleles. That clarifies all. Thank you!

**Comment by Victor-K27** [comment]
I have a question regarding this. @giulioisac @mizraelson 
We have performed TCRseq using the NEBnext mouse kit and for the analysis I tried two approaches: 

1) using MiXCR NEBNext pipeline on my raw fastq data. 
2) running the pRESTO pipeline (through Galaxy pRESTO NEBNext Immune Sequencing Kit Workflow v3.2.0) and then using the "align" and "assemble" functions of MiXCR for the aligning of the processed reads and frequency calculation. 

Comparing the data however, it differs significantly. 
From a first look, you lose the UMI information with the second approach, which I think is due to the "generic-amplicon" preset of the align function that ignores that information

Also, only a small number of the sequences in the final report are different, but the frequency of them is vastly different. 

In general, NEBnext reccomends the pre-processing of the data to be done with pRESTO, but I do not understand what the advantage of it relative to miXCR is. 

Would you be able to provide some help?

Thanks for your time,
Victor

**Reply by mizraelson** [reply]
Hi Victor,
I don’t think there is any advantage to preprocessing the data before running MiXCR. The NEBNext preset is specifically designed to work with raw data. As you correctly mentioned, the UMI information is lost in the second approach, which can potentially lower the error correction efficiency by MiXCR and reduce the effectiveness of filtering artificial clones. While this may not significantly impact the top represented clones in the repertoire, singletons will be more reliable if the NEBNext preset is used.

Sincerely,
Mark
